data-serving/data-service/schemas/cases.schema.json

```diff
@@ -9,6 +9,21 @@
       "__v": {
         "bsonType": "int"
       },
+      "caseReference": {
+        "bsonType": "object",
+        "additionalProperties": false,
+        "properties": {
+          "_id": {
+            "bsonType": "objectId"
+          },
+          "dataSourceId": {
+            "bsonType": "string"
+          },
+          "dataEntryId": {
```

Comment:
nit: I'm not sure if dataSourceId and dataEntryId are the most clear names. I looked at this PR before I read your deduping doc and was confused by what they represented. Maybe "sourceId" and "sourceEntryId"? Especially for dataEntryId I didn't understand that it was the id that came from the source.

Reply:
Makes sense to me -- done.